sdcm/nemesis.py

```diff
@@ -687,6 +800,178 @@ def _modify_table_property(self, name, val, filter_out_table_with_counter=False,
         with self.tester.cql_connection_patient(self.target_node) as session:
             session.execute(cmd)
 
+    @staticmethod
+    def _random_column_type(column_types_selected=None, max_level=1, avoid_types=None):
+        if column_types_selected is None:
+            column_types_selected = {}
+        if avoid_types is None:
+            avoid_types = []
+        if max_level == 0:
+            excluded_types = set(avoid_types + list(column_types_selected.keys()) + ['set', 'list', 'map'])
+            allowed_types = [e for e in ALL_COLUMNS if e not in excluded_types]
+        else:
+            excluded_types = set([e for e in column_types_selected.keys() if e not in ['set', 'list', 'map']] + avoid_types)
+            allowed_types = [e for e in ALL_COLUMNS if e not in excluded_types]
+        while allowed_types:
+            column_type = random.choice(allowed_types)
+            if column_type in avoid_types:
+                continue
+            column_type_info = column_types_selected.get(column_type, None)
+            if column_type_info is None:
+                column_type_info = column_types_selected[column_type] = {}
+            if column_type == 'map':
+                defined_subtypes1 = column_type_info.get('subtype1', None)
+                if defined_subtypes1 is None:
+                    defined_subtypes1 = column_type_info['subtype1'] = {}
+                defined_subtypes2 = column_type_info.get('subtype2', None)
+                if defined_subtypes2 is None:
+                    defined_subtypes2 = column_type_info['subtype2'] = {}
+                subtype1 = Nemesis._random_column_type(defined_subtypes1, max_level - 1, avoid_types + COLUMNS_NOT_EMBEDDABLE)
+                subtype2 = Nemesis._random_column_type(defined_subtypes2, max_level - 1, avoid_types + COLUMNS_NOT_EMBEDDABLE)
+                if subtype1 and subtype2:
+                    return 'map<%s,%s>' % (subtype1, subtype2)
+                allowed_types.remove('map')
+                continue
+            if column_type in ['list', 'set']:
+                subtype = Nemesis._random_column_type(column_type_info, max_level - 1, avoid_types + COLUMNS_NOT_EMBEDDABLE)
+                if subtype:
+                    return '%s<%s>' % (column_type, subtype)
+                allowed_types.remove(column_type)
+                continue
+            return column_type
+        return None
```

Comment:
Looks the same as FieldTypeDefinition.get_random()?

Reply:
Fixed.